momento-sdk/src/main/java/momento/sdk/SubscriptionWrapper.java

```diff
@@ -155,15 +157,23 @@ private void handleSubscriptionItem(_SubscriptionItem item) {
 
   private void handleSubscriptionDiscontinuity(_SubscriptionItem discontinuityItem) {
     logger.debug(
-        "{}, {}, {}, {}",
+        "discontinuity {}, {}, {}, {}, {}",
         options.getCacheName(),
         options.getTopicName(),
         discontinuityItem.getDiscontinuity().getLastTopicSequence(),
-        discontinuityItem.getDiscontinuity().getNewTopicSequence());
+        discontinuityItem.getDiscontinuity().getNewTopicSequence(),
+        discontinuityItem.getDiscontinuity().getNewSequencePage());
+
+    options.onDiscontinuity(
+        new TopicDiscontinuity(
+            discontinuityItem.getDiscontinuity().getLastTopicSequence(),
+            discontinuityItem.getDiscontinuity().getNewTopicSequence(),
+            discontinuityItem.getDiscontinuity().getNewSequencePage()));
   }
```

Comment:
On a discontinuity we also have to set the subscriptionState's topic sequence number and topic sequence page to the new values from the discontinuity.

Reply:
Good catch! I think we missed that in JS SDK: https://github.com/momentohq/client-sdk-javascript/blob/5ead1273b4ab107cb2343ff302ec1fc053acbe50/packages/client-sdk-web/src/internal/pubsub-client.ts#L229-L239?